packages/web3/src/index.ts

```diff
@@ -15,6 +15,258 @@ You should have received a copy of the GNU Lesser General Public License
+ *
+ * #### Example: Remote Geth Node
+ *
+ * ```ts
+ * // Using a remote node provider, like Alchemy (https://www.alchemyapi.io/supernode), is simple.
+ * import { Web3 } from "web3";
+ * let web3: Web3 = new Web3("https://eth-mainnet.alchemyapi.io/v2/your-api-key");
+ * ```
+ *
+ * ## providers
+ *
+ * ```ts
+ * web3.providers
+ * web3.eth.providers
+ * ```
+ * Contains the current available providers.
+ *
+ * #### Value
+ *  `Object` with the following providers:
+ *
+ *
+ *  + `Object` - `HttpProvider`: HTTP provider, does not support subscriptions.
+ *
+ *
+ *  + `Object` - `WebSocketProvider`: The WebSocket provider is the standard for usage in legacy browsers.
+ *
+ *
+ *  + `Object` - `IpcProvider`: The IPC provider is used in node.js dapps when running a local node. Gives the most secure connection.
+ *
+ *
+ * #### Example
+ * ```ts
+ * import Web3 from 'web3';
+ * // use the given Provider or instantiate a new websocket provider
+ * let web3 = new Web3(Web3.givenProvider || 'ws://remotenode.com:8546');
+ * // or
+ * let web3 = new Web3(Web3.givenProvider || new Web3.providers.WebsocketProvider('ws://remotenode.com:8546'));
+ *
+ * // Using the IPC provider in node.js
+ * var web3 = new Web3(new Web3.providers.IpcProvider('/Users/myuser/Library/Ethereum/geth.ipc')); // mac os path
+ * // on windows the path is: "\\\\.\\pipe\\geth.ipc"
+ * // on linux the path is: "/users/myuser/.ethereum/geth.ipc"
+ * ```
+ * #### Configuration
+ *
+ * ```ts
+ *
+ * //===
+ * //Http
+ * //===
+ *
+ * import Web3HttpProvider, { HttpProviderOptions } from "web3-providers-http";
+ *
+ * let options: HttpProviderOptions = {
+ * 	providerOptions: {
+ * 		keepalive: true,
+ * 		credentials: "omit",
+ * 		headers: {
+ * 			"Access-Control-Allow-Origin": "*",
+ * 		},
+ * 	},
+ * };
+ *
+ *
+ * var provider = new Web3HttpProvider("http://localhost:8545", options);
+ * web3.setProvider(provider);
+ *
+ * //===
+ * //WebSockets
+ * //===
+ * import Web3WsProvider, {
+ * 	ClientOptions,
+ * 	ClientRequestArgs,
+ * 	ReconnectOptions,
+ * } from "web3-providers-ws";
+ *
+ *
+ * let clientOptions: ClientOptions = {
+ * 	// Useful for credentialed urls, e.g: ws://username:password@localhost:8546
+ * 	headers: {
+ * 		authorization: "Basic username:password",
+ * 	},
+ * 	maxPayload: 100000000,
+ * };
+ *
+ * // Enable auto reconnection
+ * let reconnectOptions: ReconnectOptions = {
+ * 	autoReconnect: true,
+ * 	delay: 5000, // ms
+ * 	maxAttempts: 5,
+ * };
+ *
+ * //clientOptions and reconnectOptions are optional
+ * //clientOptions: ClientOptions | ClientRequestArgs
+ * let ws = new Web3WsProvider(
+ * "ws://localhost:8546",
+ * clientOptions,
+ * reconnectOptions
+ * );
+ * web3.setProvider(ws);
+ *
+ * ```
+ * More information for the Http and Websocket provider modules can be found here:
+ *
+ *
+ * - {@link HttpProvider}
+ *
+ *
+ * - {@link WebSocketProvider}
+ *
+ * See details: {@link Web3.providers}
+ *
+ *
+ * ## givenProvider
+ *
```

Comment:
If you decide to give information on `return` for `web3.providers` you should give a `return` section for all attributes like givenProvider